modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java

```diff
@@ -0,0 +1,697 @@
+package org.openapitools.codegen.languages;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.media.StringSchema;
+import io.swagger.v3.oas.models.servers.Server;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.*;
+import org.openapitools.codegen.meta.features.*;
+import org.openapitools.codegen.utils.ModelUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openapitools.codegen.utils.StringUtils.*;
+import static org.openapitools.codegen.utils.StringUtils.camelize;
+
+public abstract class AbstractDartCodegen extends DefaultCodegen {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractDartCodegen.class);
+
+    public static final String PUB_LIBRARY = "pubLibrary";
+    public static final String PUB_NAME = "pubName";
+    public static final String PUB_VERSION = "pubVersion";
+    public static final String PUB_DESCRIPTION = "pubDescription";
+    public static final String PUB_AUTHOR = "pubAuthor";
+    public static final String PUB_AUTHOR_EMAIL = "pubAuthorEmail";
+    public static final String PUB_HOMEPAGE = "pubHomepage";
+    public static final String USE_ENUM_EXTENSION = "useEnumExtension";
+    public static final String SERIALIZATION_LIBRARY_CUSTOM = "custom";
```

Comment:
I would call this native

Reply:
And I would expect these in the `DartClientCodegen`

Reply:
>I would call this native

Thanks, I had hard time coming up with good name.

>And I would expect these in the DartClientCodegen

Diff is outdated, it is there now!